pkgs/applications/audio/sidplayfp/default.nix

```diff
@@ -1,20 +1,32 @@
-{ stdenv, fetchurl, pkgconfig, libsidplayfp }:
+{ stdenv, lib, fetchurl
+, pkgconfig
+, libsidplayfp
+, alsaSupport ? stdenv.isLinux, alsaLib
+, pulseSupport ? stdenv.isLinux, libpulseaudio
+}:
```

Comment:
not sure if this is needed in this instance, but it's common to do:
```
  assert alsaSupport -> alsaLib;
  assert alsaSupport -> libpulseaudio;
```

Reply:
Why exactly is this common? I don't quite get the point of it, maybe I'm missing something. If it's e.g. to protect against alsaLib being unavailable for some platform wouldn't `alsaSupport = false;` be enough for that?

Reply:
I usually see this in packages which reference the libraries in question. E.g:
```
 buildInputs = [ (some-lib.override{ enablePython=true; }) ]'
```
but it's not always guarenteed that python will be available.

Either way, I don't see a case where it will fail on linux, but might be useful for a darwin user that an assumption is broken.